source/eulerian-path.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include "common.h"
+
+struct edge {
+	int value;
+	list<edge>::iterator it;
+};
```

Comment:
`Edge` is quite likely to conflict with other things we might write during a contest, I'd move this struct inside `EulerPath`